timemachine/md/hrex.py

```diff
@@ -357,6 +355,24 @@ def get_samples_by_iter_by_replica(
     return samples_by_iter_by_replica
 
 
+def trajectories_by_replica_to_by_state(
```

Comment:
organization: currently the HREX module is generic with respect to the data type that we use for samples (which is useful for simple tests on toy systems). Since this assumes `NDArray`, I'd probably prefer to put this with the application code in `fe.free_energy`, as it's at the same level of abstraction as `extract_trajectories_by_replica` defined there.

Reply:
Moved to `fe.free_energy` in https://github.com/proteneer/timemachine/pull/1350/commits/9aa9739e74459e15b2e27704e391563a088bfd94